TourGuide/src/main/java/tourGuide/controller/TourGuideController.java

```diff
@@ -41,22 +37,45 @@ public class TourGuideController {
     ITripPricerService tripPricerService;
 
     /**
+     * Entry point
      *
-     * @return
+     * @return A welcome message
      */
     @RequestMapping("/")
     public String index() {
         return "Greetings from TourGuide!";
     }
 
     /**
+     * Get a list of every user's most recent location
      *
-     * @return
+     * @return a JSON mapping of userId to Locations like:
+     * {"userId":"17190a91-c3e3-4633-aace-92de27f807d7","location":{"latitude":-155.263092,"longitude":-30.622786}}
+     */
+    @RequestMapping("/getAllCurrentLocations")
+    public String getAllCurrentLocations() {
+        return JsonStream.serialize(gpsUtilService.getAllCurrentLocations(userService.getAllUsers()));

```

Comment:
What do we do in case of a failure? What if the provided Json is not serializable?